### PR TITLE
[ISSUE #8979] Add configurable switch for timer message retry logic 

### PR DESCRIPTION
**Pull request by chi3316**
### Which Issue(s) This PR Fixes

Fixes [#8979 ](https://github.com/apache/rocketmq/issues/8979)

### Brief Description
- Added a configuration switch `TimerEnableRetryUntilSuccess` to control retry behavior.
- When `TimerEnableRetryUntilSuccess` is disabled, a message will be retried up to 3 times before being discarded.
- Improved logging to better reflect message processing and retry status.

### CODE REVIEW

File: store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java [hunk 1]
```diff
@@ -1084,46 +1084,42 @@ public int doPut(MessageExtBrokerInner message, boolean roll) throws Exception {
...
+                        return PUT_NEED_RETRY;
+                    }
             }
-            LOGGER.warn("Retrying to do put timer msg retryNum:{} putRes:{} msg:{}", retryNum, putMessageResult, message);
         }
-        return PUT_NO_RETRY;
+        return PUT_NEED_RETRY;
     }
 
     public MessageExtBrokerInner convertMessage(MessageExt msgExt, boolean needRoll) {
```
Comment (RongtongJin) [inline store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java:1125]: It would be beneficial to include a unit test

Comment (chi3316) [reply store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java:1127]: I’ve added  a unit test to cover this function. Please take a look👀

**Review by RongtongJin — APPROVED** [review]
LGTM

**Review by guyinyou — APPROVED** [review]
lgtm

**Review by GenerousMan — APPROVED** [review]
LGTM

**Pull request merged**